snapcraft/_store.py

```diff
@@ -100,6 +101,119 @@ def _requires_login():
+        iter(cmd_list)
+    except TypeError:
+        raise
+    not_found_err = not_found_err or EnvironmentError(
+        '`{}` not found.'.format(cmd_list[0]))
+    failure_err = failure_err or EnvironmentError(
+        '`{}` failed.'.format(command))
+    try:
+        subprocess.check_call(
+            cmd_list, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    except FileNotFoundError as err:
+        raise not_found_err
+    except subprocess.CalledProcessError as err:
+        raise failure_err
+    yield
+
+
+@contextmanager
+def _requires_path_exists(path, error=None):
+    if error is None:
+        error_message = 'Required path does not exist: {}'.format(path)
+    if not os.path.exists(path):
+        raise EnvironmentError(error_message)
+    yield
+
+
+def _travis_encrypt_config(config_path):
+    """Encrypt given snapcraft config file for Travis jobs.
+
+    Depends on working `travis` CLI (boo!)
+    """
+    cmd = [
+        'travis', 'encrypt-file',
+        '--add', '--force', '--decrypt-to', '.snapcraft.cfg',
+        config_path, '.snapcraft.cfg.enc',
+    ]
+    try:
+        subprocess.check_output(cmd, stderr=subprocess.PIPE)
+    except subprocess.CalledProcessError as err:
+        raise RuntimeError(
+            '`travis encrypt-file` failed: {}\n{}'.format(
+                err.returncode, err.stderr.decode()))
+
+
+@_requires_command_success(
+    'travis version',
+    EnvironmentError(
+        'Travis CLI (`travis`) is not available.\n'
+        'Please install it (e.g. `sudo gem install travis`) '
+        'before trying this again.'),
+    EnvironmentError(
+        'Travis CLI (`travis version`) is not functional.\n'
+        'Make sure it works correctly in your system '
+        'before trying this again.'))
+@_requires_command_success(
+    'git status',
+    EnvironmentError(
+        'Git (`git`) is not available, this tool cannot verify '
+        'its prerequisites.\n'
+        'Please install it (e.g. `sudo apt-get install git`) '
+        'before trying this again.'),
+    EnvironmentError(
+        'The current directory is not a Git repository.\n'
+        'Please switch to the desired project repository where '
+        'Travis should be enabled.'))
+@_requires_path_exists(
+    '.travis.yml',
+    'Travis project is not initialised for the current directory.\n'
+    'Please initialise Travis project (e.g. `travis init`) with '
+    'appropriate parameters.')
+def enable_travis(snap_name):
+    logger.info(
+        'Enabling this project for using snapcraft in Travis testbeds.'
+    )
+
+    series = storeapi.constants.DEFAULT_SERIES
+    # XXX cprov 20160923: Using unresolved (name, series) caveat,
+    # instead of 'snap-id'. Mixed feelings on why it is supported.
+    # Needs caveat syntax for restricting store request origin (IP
+    # or reverse-dns).
+    packages = [{'name': snap_name, 'series': series}]
+
+    logger.info('Acquiring specific authorisation information ...')
```

Comment:
nit: you use "authorization" below. I prefer it with z, but I'm not sure which one is more correct.
Also above you are using initialised a couple of times. I would spell that with a z too.